ground/gcs/src/plugins/config/configautotunewidget.cpp

```diff
@@ -367,14 +369,30 @@ void ConfigAutotuneWidget::recomputeStabilization()
     // critically damped;
     const double zeta_o = 1.3;
     const double kp_o = 1 / 4.0 / (zeta_o * zeta_o) / (1/wn);
+    const double ki_o = 0.75 * kp_o / (2 * M_PI * tau * 10.0);
 
     // For now just run over roll and pitch
-    for (int i = 0; i < 2; i++) {
+    for (int i = 0; i < 3; i++) {
         double beta = exp(systemIdentData.Beta[i]);
 
-        double ki = a * b * wn * wn * tau * tau_d / beta;
-        double kp = tau * tau_d * ((a+b)*wn*wn + 2*a*b*damp*wn) / beta - ki*tau_d;
-        double kd = (tau * tau_d * (a*b + wn*wn + (a+b)*2*damp*wn) - 1) / beta - kp * tau_d;
+        double ki;
+        double kp;
+        double kd;
+
+        switch (i) {
+        case 0: // Roll
+        case 1: // Pitch
+            ki = a * b * wn * wn * tau * tau_d / beta;
+            kp = tau * tau_d * ((a+b)*wn*wn + 2*a*b*damp*wn) / beta - ki*tau_d;
+            kd = (tau * tau_d * (a*b + wn*wn + (a+b)*2*damp*wn) - 1) / beta - kp * tau_d;
+            break;
+        case 2: // Yaw
+            beta = exp(0.6 * (systemIdentData.Beta[SystemIdent::BETA_PITCH] - systemIdentData.Beta[SystemIdent::BETA_YAW]));
```

Comment:
This is supposed to be -;  it basically looks at the gain difference observed between pitch and yaw, and picks a value inbetween to scale it.  It doesn't scale things by the full difference, because actuators are nonlinear and it's better to have yaw be undertuned a little than couple badly into the pitch and roll axes.